### PR TITLE
added Alcohol class

### PR DESCRIPTION
**Pull request by BarTes8**

### CODE REVIEW

**Comment by BarTes8** [comment]
.

File: shm/inc/Alcohol.hpp [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+class Alcohol : public Cargo {
+public:
+    Alcohol(const std::string&, size_t, size_t, size_t);
+    ~Alcohol() override;
+
+    size_t getPrice() const override; { return basePrice_ * percentage_ / 96; }
+    std::string getName() const override { return name_; }
+    size_t getAmount() const override { return amount_; }
+    size_t getBestPrice() const override { return basePrice_; }
```
Comment (MoreThanADC) [inline shm/inc/Alcohol.hpp]: Change getBestPrice to getBasePrice xD

File: shm/inc/Alcohol.hpp [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+#include "Cargo.hpp"
+
+constexpr size_t basePrice = 96;
+
+class Alcohol : public Cargo {
+public:
+    Alcohol(const std::string&, size_t, size_t, size_t);
+    ~Alcohol() override;
+
+    size_t getPrice() const override; { return basePrice_ * percentage_ / 96; }
```
Comment (MoreThanADC) [inline shm/inc/Alcohol.hpp]: Delete ";" after override

File: shm/inc/Alcohol.hpp [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    std::string getName() const override { return name_; }
+    size_t getAmount() const override { return amount_; }
+    size_t getBestPrice() const override { return basePrice_; }
+
```
Comment (MoreThanADC) [inline shm/inc/Alcohol.hpp]: Add a comment after which class you are overwriting

File: shm/inc/Alcohol.hpp [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    size_t getPercentage() const { return percentage_; }
+
+    void setBasePrice(size_t basePrice) { basePrice_ = basePrice; } 
+
+private:
+    size_t percentage;
+};
```
Comment (MoreThanADC) [inline shm/inc/Alcohol.hpp]: add space after last line (enter)

File: shm/CMakeLists.txt [hunk 5]
```diff
@@ -13,6 +13,7 @@ set(INCLUDE_DIR ${CMAKE_SOURCE_DIR}/..)
 set(SOURCE_DIR ${CMAKE_SOURCE_DIR}/src)
 
 set(SOURCES
+    ${SOURCE_DIR}/Alcohol.cpp
     ${SOURCE_DIR}/Cargo.cpp
```
Comment (MoreThanADC) [inline shm/CMakeLists.txt:17]: delete Cargo.cpp,  this file no longer exists

**Review by MoreThanADC — CHANGES_REQUESTED** [review]
Fix a few little things

File: shm/inc/Alcohol.hpp [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+#include <string>
+
+#include "Cargo.hpp"
+
+constexpr size_t basePrice = 96;
+
+class Alcohol : public Cargo {
+public:
+    Alcohol(const std::string&, size_t, size_t, size_t);
```
Comment (mariuszlisowski) [inline shm/inc/Alcohol.hpp]: variable names

File: shm/inc/Alcohol.hpp [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+    std::string getName() const override { return name_; }
+    size_t getAmount() const override { return amount_; }
+    size_t getBasePrice() const override { return basePrice_; }
+
+    size_t getPercentage() const { return percentage_; }
+
+    void setBasePrice(size_t basePrice) { basePrice_ = basePrice; } 
+
+private:
+    size_t percentage_;
```
Comment (mariuszlisowski) [inline shm/inc/Alcohol.hpp:28]: const?

File: shm/inc/Alcohol.hpp [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+
+constexpr size_t basePrice = 96;
+
+class Alcohol : public Cargo {
+public:
+    Alcohol(const std::string&, size_t, size_t, size_t);
+    ~Alcohol() override;
+
+    //override Cargo
+    size_t getPrice() const override { return basePrice_ * percentage_ / 96; }
```
Comment (mariuszlisowski) [inline shm/inc/Alcohol.hpp:15]: magic number

File: shm/inc/Alcohol.hpp [hunk 9]
```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include <string>
+
+#include "Cargo.hpp"
+
+constexpr size_t basePrice = 96;
```
Comment (mariuszlisowski) [inline shm/inc/Alcohol.hpp:7]: not MAX_PERCENTAGE instead?

File: shm/inc/Alcohol.hpp [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+
+    //override Cargo
+    size_t getPrice() const override { return basePrice_ * percentage_ / 96; }
+    std::string getName() const override { return name_; }
+    size_t getAmount() const override { return amount_; }
+    size_t getBasePrice() const override { return basePrice_; }
+
+    size_t getPercentage() const { return percentage_; }
+
+    void setBasePrice(size_t basePrice) { basePrice_ = basePrice; }
```
Comment (mariuszlisowski) [inline shm/inc/Alcohol.hpp:25]: YAGNI (You Aren't Gonna Need It) - basePrice is already set in a constructor.

File: shm/src/Alcohol.cpp [hunk 11]
```diff
@@ -0,0 +1,10 @@
+#include "shm/inc/Alcohol.hpp"
+
+Alcohol::Alcohol(const std::string& name, size_t amount, size_t basePrice, size_t percentage) 
+    : name_(name)
+    , amount_(amount)
+    , basePrice_(basePrice)
+    , percentage_(percentage)
+{}
+
+Alcohol::~Alcohol() override = default;
```
Comment (mariuszlisowski) [inline shm/src/Alcohol.cpp:8]: keyword 'override' not allowed here

**Review by mariuszlisowski — CHANGES_REQUESTED** [review]

**Pull request closed without merging**